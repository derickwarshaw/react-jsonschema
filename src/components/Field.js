import React, { Component } from 'react';
import PropTypes from 'prop-types';
import TextField from './TextField';

class Field extends Component {
  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(e) {
    const { schema, path } = this.props;
    this.props.onChange({
      schema,
      path,
      value: e.target.value,
    });
  }

  render() {
    const { path, schema, formData } = this.props;
    return (
      <TextField
        schema={schema}
        formData={formData}
        name={path}
        label={schema.title}
        onChange={this.handleChange}
      />
    );
  }
}

Field.propTypes = {
  path: PropTypes.string,
  schema: PropTypes.object,
  formData: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.bool,
  ]),
  onChange: PropTypes.func, // { path, schema, value }
};

export default Field;
