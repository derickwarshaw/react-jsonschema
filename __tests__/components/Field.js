import React from 'react';
import renderer from 'react-test-renderer';
import types from '../../src/constants/types';
import Field from '../../src/components/Field';

describe('<Field />', () => {
  it('should render a <Field /> component', () => {
    const path = 'formData';
    const schema = { title: 'Name', type: types.STRING };
    const formData = 'Vu Tran';
    const component = renderer.create(
      <Field
        path={path}
        schema={schema}
        formData={formData}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
