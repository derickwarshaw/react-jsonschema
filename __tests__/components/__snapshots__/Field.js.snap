// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Field /> should render a <Field /> component 1`] = `
<div
  data-css-vbwgzr=""
>
  <label
    data-css-1luuo4y=""
    htmlFor="formData"
  >
    Name
  </label>
  <input
    data-css-14dfslc=""
    name="formData"
    onChange={[Function]}
    type="text"
    value="Vu Tran"
  />
</div>
`;
